makefile

```diff
@@ -7,6 +7,10 @@ ifdef codecov
   LIBS    += -lgcov
 endif
 
+ifdef ugrid_debug
```

Comment:
@jeff-cohere With this change, I wanted to allow for the addition of a compiler flag `-DUGRID_DEBUG`, but here is the issue:

- `make ugrid=1 V=1`: The new compiler flag isn't used
- `make ugrid=1 V=1 all-gmake`: The new compiler flag is used

Any what is going wrong?

Reply:
I'm not sure. It might be related to the fact that the first command invokes the second indirectly, but maybe doesn't capture the `ugrid=1` flag you're passing. If that's true,  we need to explicitly pass along arguments from the default (`all`) target to `all-gmake`.

More broadly: this build system has worked for us for a while now, but it's still pretty confusing (we have all of the different spellings of makefiles, for example), and it's not clear to a newcomer what gets invoked where unless said newcomer is versed in all of Make's various assumptions (which fewer and fewer people are as time passes).

As we move toward becoming part of a larger ecosystem, we might need a more "turnkey" build system that is a little easier for a non-expert to extend.

Reply:
Yeah, I'm not complaining! This build system has taken the project pretty far, and I appreciate not having had to touch it very much. Build systems are difficult in general, and it's always difficult to strike a balance between dealing with existing quirks and spending time improving the system itself.

Reply:
Note that gmake is quite a bit simpler if you always build from the top level because it sees all the dependencies. It's clutter and harder to test if you want stand-alone makefiles in each subdirectory. Sometimes people want subdirectories to be stand-alone because the idea is that a user would copy them somewhere else as step one of a new project. If we need to refactor/change the build, it would be useful to gather requirements.